share/completions/reg.fish

```diff
@@ -1,4 +1,20 @@
+function __reg_run_reg_safely
+    if which reg > /dev/null
```

Comment:
1. `which` is a cheesy external command that might not be installed and shouldn't be used
2. When `which` doesn't find a command it prints an *error* to stderr, so this spews error messages
3. We already only load the description when the command to be completed is an executable thing

Concretely, this prevents me from testing this via `function reg; wine reg $argv; end`, even tho that would work.

Just remove the entire check. If you *have* to have a check, use `type -q reg`. Just `if type -q reg`. That checks if `reg` is any kind of runnable thing (function, command, builtin).